src/openms/include/OpenMS/KERNEL/MSSpectrum.h

```diff
@@ -330,6 +330,78 @@ namespace OpenMS
       return integer_data_arrays_;
     }
 
+    /// Returns an iterator to the first float data array with the given name. 
+    /// The End iterator is returned in case no data array with given name exists.
+    typename FloatDataArrays::iterator getFloatDataArrayByName(const String& name)
+    {
+      typename FloatDataArrays::iterator it = float_data_arrays_.begin();
+      for (; it != float_data_arrays_.end(); ++it)
+      {
+        if (it->getName() == name) return it;
+      }
+      return it;
+    }
+
+    /// Returns an iterator to the first string data array with the given name. 
+    /// The End iterator is returned in case no data array with given name exists.
+    typename StringDataArrays::iterator getStringDataArrayByName(const String& name)
```

Comment:
the `end` iterator is not such a bad idea, I think better than anything else. Otherwise we would have to return an empty (static) array but it would be very bad if somebody were to try and insert things in there ...